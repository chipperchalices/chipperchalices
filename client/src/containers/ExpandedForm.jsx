import React, {Component} from 'react';
import axios from 'axios';
import moment from 'moment';
import TextField from 'material-ui/TextField';
import {Dialog, Avatar} from 'material-ui';
import {IconButton, FontIcon} from 'material-ui';
import {DatePicker, TimePicker} from 'material-ui';
import AllOut from 'material-ui/svg-icons/action/all-out';
import { Button, Header, Image, Modal } from 'semantic-ui-react'

export default class ExpandedModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      job: this.props.job,
      status: true
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState({
      open: !this.state.open
    });
  }

  render() {

    return (
      <Modal trigger={<Button size='mini' floated='right' color='blue' circular icon='expand' onClick={this.handleClick}/>}
        open={this.state.open}
        onClose={this.handleClick}
        size='small'
      >
        <Modal.Header>{this.state.job.position} at {this.state.job.company.name}</Modal.Header>
        <Modal.Content image>
          <Image wrapped size='medium' src={this.state.job.company.logo_url} />
          <Modal.Description>
            <Header>Recruiter: {this.state.job.recruiter_name}</Header>
            <p>Recruiter Email: {this.state.job.recruiter_email}</p>
            <p>Application: <a href={this.state.job.position_url}>{this.state.job.position_url}</a></p>
            <p>Notes: {this.state.job.notes}</p>
          </Modal.Description>
        </Modal.Content>
      </Modal>
    );
  }
}
