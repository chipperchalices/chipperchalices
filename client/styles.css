#newApp {
	body {
	    background-color: lightblue;
	}
}

#stage2 {
	body {
	    background-color: lightblue;
	}
}

#stage3 {
	body {
	    background-color: lightblue;
	}
}

#stage4 {
	body {
	    background-color: lightblue;
	}
}

#stage5 {
	body {
	    background-color: lightblue;
	}
}

#stage6 {
	body {
	    background-color: lightblue;
	}
}


