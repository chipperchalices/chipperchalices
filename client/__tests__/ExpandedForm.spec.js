import React from 'react';
import expect from 'expect';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import ExpandedForm from '../src/containers/ExpandedForm.jsx';

//
import fakeData from '../src/dummy_data.js';
import { lightBaseTheme, MuiThemeProvider, getMuiTheme } from 'material-ui/styles';

//Shallow Render to reduce snapshot output
describe('Expanded Form Screenshot', () => {
  it('should render using fakeData (enzyme)', () => {
    const wrapper = shallow(
      <MuiThemeProvider>
        <ExpandedForm job={fakeData[0]}/>
      </MuiThemeProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

// describe('Expandedform State Should Render Items', () => {
  
//   let form;
//   const muiTheme = getMuiTheme();
//   beforeEach(() => {
//     // mount = createMount();
//     form = mount(<MuiThemeProvider><ExpandedForm job={fakeData[0]}/></MuiThemeProvider>);
//     form = form.find(EditForm);
//   });
//   // afterEach(() => {
//   //   mount.cleanUp();
//   // });

//   it('should call componentDidMount once', () => {
    
//     sinon.spy(EditForm.prototype, 'componentDidMount');
//     expect(EditForm.prototype.componentDidMount.calledOnce).toEqual(true);
//   });

//   it('should not be open when launched', () => {
//     expect(form.state().open).toEqual(false);
//   });
// });

describe('Expanded Form Props', () => {
  let form;

  beforeEach(()=> {
    form = shallow(<MuiThemeProvider><ExpandedForm job={fakeData[0]}/></MuiThemeProvider>);
  });
  
  it('should Receive All Props', () => {
    expect(Object.keys(form.props().job).length).toEqual(9);
  });

  it('should have position', () => {
    expect(form.props().job.position).toEqual('UI Developer');
  });

  it('should have position id', () => {
    expect(form.props().job.id).toEqual(1);
  });

  it('should have company object', () => {
    expect(Object.keys(form.props().job.company).length).toEqual(8);
  });

  it('should have company name', () => {
    expect(form.props().job.company.name).toEqual('Vlocity');
  });

  it('should have company id', () => {
    expect(form.props().job.company.id).toEqual(12);
  });

  it('should have date', () => {
    expect(form.props().job.statusDate).toEqual('2017-09-02');
  });

  it('should have position url', () => {
    expect(form.props().job.positionUrl).toEqual('https://boards.greenhouse.io/vlocity/jobs/781103');
  });

  it('should have recruiter name', () => {
    expect(form.props().job.recruiterName).toEqual('Jane Smith');
  });

  it('should have recruiter email', () => {
    expect(form.props().job.recruiterEmail).toEqual('123@gmail.com');
  });

});

describe('Expanded Form', () => {
  
  let form;
  const muiTheme = getMuiTheme();
  beforeEach(() => {
    form = mount(<ExpandedForm job={fakeData[0]}/>, {
      context: {muiTheme},
      childContextTypes: {muiTheme: React.PropTypes.object}
    });
  });

  it('should allow to open and close form', () => {
    expect(form.state().open).toEqual(false);
    form.find('IconButton').find('button').simulate('click');
    expect(form.state().open).toEqual(true);
    form.find('IconButton').find('button').simulate('click');
    expect(form.state().open).toEqual(false);
  });

  describe('Expanded form handleclick functions', () => {

    let form;
    const muiTheme = getMuiTheme();

    beforeEach(() => {
      form = mount(<ExpandedForm job={fakeData[0]}/>, {
        context: {muiTheme},
        childContextTypes: {muiTheme: React.PropTypes.object}
      });
    });

    it('should change state to be clicked', () => {
      expect(form.state().open).toEqual(false);
      
      form.instance().handleClick();
  
      expect(form.state().open).toEqual(true);
    });

  });
});
