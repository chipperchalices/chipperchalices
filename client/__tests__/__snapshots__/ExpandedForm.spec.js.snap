// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expanded Form Screenshot should render using fakeData (enzyme) 1`] = `
<ExpandedModal
  job={
    Object {
      "company": Object {
        "companyUrl": "www.vlocity.com",
        "description": "Vlocity is a leading industry cloud software company that delivers award winning industry-specific applications on the Salesforce platform.",
        "id": 12,
        "industry": "Internet",
        "location": "San Francisco, CA",
        "logoUrl": "https://tinyurl.com/ya54jrkq",
        "name": "Vlocity",
        "ratings": "3.8",
      },
      "currentStatus": "Interested",
      "id": 1,
      "notes": null,
      "position": "UI Developer",
      "positionUrl": "https://boards.greenhouse.io/vlocity/jobs/781103",
      "recruiterEmail": "123@gmail.com",
      "recruiterName": "Jane Smith",
      "statusDate": "2017-09-02",
    }
  }
/>
`;
